workflow/validate/validate.go

```diff
@@ -198,19 +204,16 @@ func ValidateWorkflow(wftmplGetter templateresolution.WorkflowTemplateNamespaced
 	return wfConditions, nil
 }
 
-// ValidateWorkflow accepts a workflow template and performs validation against it.
-func ValidateWorkflowTemplate(wftmplGetter templateresolution.WorkflowTemplateNamespacedGetter, cwftmplGetter templateresolution.ClusterWorkflowTemplateGetter, wftmpl wfv1.TemplateHolder) error {
-	ctx := newTemplateValidationCtx(nil, ValidateOpts{})
-	tmplCtx := templateresolution.NewContext(wftmplGetter, cwftmplGetter, wftmpl, nil)
+// ValidateWorkflowTemplate accepts a workflow template and performs validation against it.
+func ValidateWorkflowTemplate(wftmplGetter templateresolution.WorkflowTemplateNamespacedGetter, cwftmplGetter templateresolution.ClusterWorkflowTemplateGetter, wftmpl *wfv1.WorkflowTemplate) (*wfv1.WorkflowConditions, error) {
+	wf := common.ConvertWorkflowTemplateToWorkflow(wftmpl)
+	return ValidateWorkflow(wftmplGetter, cwftmplGetter, wf, ValidateOpts{IgnoreEntryPoint: true})
```

Comment:
The `WorkflowTemplate` may actually have an `entrypoint`, in which case we might actually want to do validation with the `entrypoint` accounted for. Why not make the `IgnoreEntrypoint` flag dependent on whether or not the `WorkflowTemplate` actually has one?

Reply:
Entrypoint will be optional during `WorkflowTemplate` creation time. But It will be validated during the submittable as `Workflow` in workflow validation.

Reply:
Sure, but if the `entrypoint` is provided, then we should do a full validation during `WorkflowTemplate` creation time. It would be a bad user experience to have an invalid `WorkflowTemplate` accepted during creation time and then rejected during execution time later on.